### PR TITLE
Improving draw method

### PR DESCRIPTION
**Pull request by anushkrishnav**
First-time contributor here !! Pretty interesting projects 

I have worked on fixing the bug: Circuit. draw() returns an error when the gate parameter is a string
It was a straightforward task and didn't take a lot of time, I have also written the code with spend in mine, For a 100 gate it takes less than a second to draw the output, I have not written a test or doc wanted to get the teams approval after which will finish up with test cases for the draw and _string_to_sympy methods.

The attached collab will give an example of the newer code in motion, I have created a random gate generator that generates n gates( currently set to 100 and n_qubits =2) 
Link: 

Fix for Issue: #388 

Colab: https://colab.research.google.com/drive/1g4ll38cIn-27-tB4KCYNzV6gp6MK-Bzk?usp=sharing

### CODE REVIEW

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
Good stuff ! The results look good and if you make these small changes and the automated tests pass, we're good !
I hope this first contribution is a good time so far

The automated tests currently fail because our linter, pycodetest, sees a few things

https://github.com/goodchemistryco/Tangelo/actions/runs/9281488084/job/25537568509?pr=391

```
../tangelo/linq/circuit.py:34:1: E303 too many blank lines (3)
../tangelo/linq/circuit.py:192:1: W293 blank line contains whitespace
../tangelo/linq/circuit.py:214:5: E303 too many blank lines (2)
- generated xml file: /home/runner/work/Tangelo/Tangelo/dev_tools/junit/pycodestyle-test-results.xml -
=========================== short test summary info ============================
FAILED test_conformance.py::TestCodeFormat::test_conformance - AssertionError: 3 != 0 : Found code style errors and warnings.
```

File: tangelo/linq/circuit.py [hunk 1]
```diff
@@ -179,14 +181,35 @@ def applied_gates(self):
         """
         return self._applied_gates if "CMEASURE" in self.counts else self._gates
 
-    def draw(self):
+    def old_draw(self):
```
[the remark this answers is not in the document]
Comment (anushkrishnav) [reply tangelo/linq/circuit.py:184]: Awesome will finish the docstring and test in a few hrs and send the final commit

**Comment by anushkrishnav** [comment]
> ../tangelo/linq/circuit.py:192:1: W293 blank line contains whitespace

Had a good time working on the PR, its an interesting projects will keep working on the other issues on board as well. I am aware of the lining issue, wanted to know if i was in the right direction , will finish up the code now

File: tangelo/linq/circuit.py [hunk 2]
```diff
@@ -179,14 +181,35 @@ def applied_gates(self):
...
         # circular import
         from tangelo.linq.translator.translate_cirq import translate_c_to_cirq
         cirq_circ = translate_c_to_cirq(self)
         # Remove identity gates that are added in translate_c_to_cirq (to ensure all qubits are initialized) before drawing.
         cirq_circ.__delitem__(0)
         return SVGCircuit(cirq_circ)
+    
+    def draw(self):
+        from tangelo.linq.translator.translate_cirq import translate_c_to_cirq
+        circuit_copy = copy.deepcopy(self)
```
[the remark this answers is not in the document]
Comment (anushkrishnav) [reply tangelo/linq/circuit.py:195]: Got it will use the exsisting ones

**Review by ValentinS4t1qbit — APPROVED** [review]
Looks good to me !

**Comment by ValentinS4t1qbit** [comment]
Thank you so much for your contributions @anushkrishnav ! I hope this has been an interesting experience, and that you have a learnt a few new things :)

**Comment by anushkrishnav** [comment]
I have left a star thanks for the support, will start working on the next issue, is there anything needed from my side for the unitary hacks ?

**Pull request merged**